codec/std/codec.go

```diff
@@ -180,6 +180,10 @@ func (c *Codec) UnmarshalProposal(bz []byte) (gov.Proposal, error) {
 	}, nil
 }
 
+func (c *Codec) NewMsgSubmitProposalI() gov.MsgSubmitProposalI {
```

Comment:
I can't recall if we discussed this, but this API doesn't sit right with me. A codec shouldn't do this. We should be able to just call the constructor.

Reply:
I think we did discuss this briefly in our last call. My original design was to just pass a `ctr func() MsgSubmitProposalI` constructor argument around. But I thought this approach was more aligned with the `Codec` approach. Are you fine with just passing a constructor func around?

Reply:
Yeah I'd opt for passing it around or coming up with another interface, but that too would have to be passed around, so might as well just go with `ctr func() MsgSubmitProposalI`.

Reply:
👍 I'll refactor